CMakeLists.txt

```diff
@@ -14,7 +14,17 @@ set(CAFFE_TARGET_VERSION "1.0.0-rc3")
 set(CAFFE_TARGET_SOVERSION "1.0.0-rc3")
 add_definitions(-DCAFFE_VERSION=${CAFFE_TARGET_VERSION})
 
+# definition needed to include CMake generated files
+add_definitions(-DCMAKE_BUILD)
```

Comment:
This is needed for static linking. Otherwise, export.hpp will skip include_symbols.hpp, which consequently miss the required .obj. This won't be an issue in linking time, but it will be an issue in runtime. You will get an error like this:

F0509 14:41:28.143733 47048 layer_factory.cpp:62] Check failed: registry.count(type) == 1 (0 vs. 1)
Unknown layer type: Input (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, Bia
s, Concat, ContrastiveLoss, Convolution, Crop, Data, Deconvolution, Dropout, DummyData, ELU, Eltwise
, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, ImageData, In
fogainLoss, InnerProduct, LRN, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Pooling, Power,
 ReLU, Reduction, Reshape, SPP, Scale, Sigmoid, SigmoidCrossEntropyLoss, Slice, Softmax, SoftmaxWith
Loss, Split, TanH, Threshold, Tile)